lib/gollum/public/gollum/javascript/editor/gollum.editor.js

```diff
@@ -306,6 +306,11 @@
               formData = new FormData();
           formData.append('file', file);
           
+          if (Object.prototype.toString.call(file) != '[object File]') {
```

Comment:
This all seems sensible to me, but wouldn't be easier to move the 'return' call all the way to the top of this function? That way we wouldn't even have to add/remove the `uploading` class.

Reply:
We need the `e.dataTransfer.files[0]` first before we can check if the uploaded object is a file. And moving that call to before we add the `uploading` class would mean that the spinner wouldn't be shown during upload, only after the upload is complete.

Reply:
Fair enough!